solr/core/src/java/org/apache/solr/request/SolrRequestInfo.java

```diff
@@ -38,7 +40,13 @@
 
 
 public class SolrRequestInfo {
-  protected final static ThreadLocal<SolrRequestInfo> threadLocal = new ThreadLocal<>();
+
+  protected final static int capacity = 150;
```

Comment:
This should be MAX_STACK_SIZE I think.  
Can anyone ( @mkhludnev ?) venture to guess how, realistically, we might reach upwards of 150?  I can't imagine more than a few, let alone 150.

Reply:
Initially I set it to 100, then there was one test (CloudAuthStreamTest) was failing due to this capacity. So increased it to 150.

Reply:
I suggest debugging/investigating there to understand _why_.  Maybe this cap has exposed an actual problem?